Use str::strip* in bootstrap

tesuji opened the pull request:
This is technically a breaking change, replacing the use of `trim_start_matches` with `strip_prefix`. However, because in `rustc -Vv` output there are no lines starting with multiple "release:", this should go unnoticed in practice.

Mark-Simulacrum commented on src/bootstrap/lib.rs:
```diff
@@ -436,10 +436,9 @@ impl Build {
+            .filter_map(|x| x.strip_prefix("release:"))
             .next()
             .unwrap()
-            .trim_start_matches("release:")
```
Can you make note of this slight behavior change in the commit description? That'll help track it down if it causes any bugs (though I don't think it should).

lzutao replied:
I added a rough explanation.

Mark-Simulacrum reviewed (commented):
r=me with commit edited

tesuji committed:
Use str::strip* in bootstrap

This commit replaces the use of `trim_start_matches`
because in `rustc -Vv` output there are no lines
starting with multiple "release:".

lzutao commented:
I added simple explanation in the commit message.

Mark-Simulacrum commented:
@bors r+ rollup

bors commented:
:pushpin: Commit 481988b0832de4b28fd1f4a52f0df2801f62bda8 has been approved by `Mark-Simulacrum`

<!-- @bors r=Mark-Simulacrum 481988b0832de4b28fd1f4a52f0df2801f62bda8 -->
<!-- homu: {"type":"Approved","sha":"481988b0832de4b28fd1f4a52f0df2801f62bda8","approver":"Mark-Simulacrum"} -->

lzutao commented:
Honestly I don't see how this would be a breaking change as I only change internal bootstrap crate .
But if you want to track down possible regressions, adding a note is fine.

Mark-Simulacrum commented:
It's not one for downstream users (e.g., people using rustup) but it could be one for distros or rustc developers.
